Dockerfile

```diff
@@ -0,0 +1,105 @@
+# syntax=docker/dockerfile:1.4
+
+# Build image to compile all packages
+# The target repository is  https://github.com/NVIDIA/doca-sosreport
+FROM ubuntu:24.04 AS build
+
+ARG PYTHON_VERSION=3.12
+
+RUN apt-get update -qy && apt-get install -qyy \
+    -o APT::Install-Recommends=false \
+    -o APT::Install-Suggests=false \
+    build-essential \
+    ca-certificates \
+    curl \
+    python3-setuptools \
+    python${PYTHON_VERSION}-dev && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+COPY --from=ghcr.io/astral-sh/uv:0.6 /uv /usr/local/bin/uv
+
+ENV UV_LINK_MODE=copy \
+    UV_COMPILE_BYTECODE=1 \
+    UV_PYTHON_DOWNLOADS=never \
+    UV_PYTHON=python3.12
+
+ENV VIRTUAL_ENV=/opt/venv \
+    PATH="/opt/venv/bin:$PATH"
+
+COPY setup.py .
```

Comment:
is there an option to copy everything in order to avoid issues in the future? with this change we have now three places that copy files, RPM, dev and now this

Reply:
we could copy the whole directory, but I'd like to avoid it. The image is already more than 1GB as is.